oauth-common/src/test/java/io/strimzi/kafka/oauth/jsonpath/CustomCheckTest.java

```diff
@@ -0,0 +1,222 @@
+        "@.missing && @.missing nin [1,2,3]", "false",
+        "\"kafka-user\" in @.['roles'].['client-roles'].['kafka']", "true",
+        "@.roles.client-roles.kafka && \"kafka-admin\" nin @.roles.client-roles.kafka", "true"
+    };
+
+    /*
+     * These are queries supported by Jayway implementation but not supported by our implementation.
+     * It's mostly the lack of whitespace.
+     */
+    final String[] poorWhitespaceQueries = {
+        "@.custom-level!=-1", "true",
+        "@.iat==null", "false",
+        "@.custom=~/CUSTOM-.+/i", "true",
+        "@.custom=='custom-value' && @.exp>1000", "false",
+        "@.custom=='custom-value'&&@.exp>1000", "false"
+    };
+
+    /*
+     * These queries return a different result in Jayway JsonPath implementation
+     *
+     * In the case of 'anyof' and 'noneof' Jayway seems to silently fail the evaluation when the referenced
+     * attribute is missing, which results in EXPRESSION and !(EXPRESSION) to both effectively evaluate to false.
+     */
+    final String[] buggyJaywayQueries = {
+        "!(@.missing anyof [null, 'username'])", "true",
+        "!(@.missing noneof [null, 'username'])", "true"
+    };
+
+    final String[] nonsensicalQueries = {
+        "['exp'] > 1000", "false",
+        "@", "true",
+        "@.custom in null", "false",
+        "'lala' > 'lala'", "false",
+        "'lala' <= null", "false",
+        "@.attr > null", "false",
+        "1 < @.attr", "false"
+    };
+
+    @Test
+    public void testJsonPathFilterQuery() throws Exception {
+
+        String[] errQueries = {
+            "@.attr == and !('admin' in @.roles)",
+            "@.exp < 'cant compare makes it false"
+        };
+
+        JsonNode json = JsonSerialization.readValue(jsonString, JsonNode.class);
+
+        testQueries(json, queries);
+        testQueries(json, poorWhitespaceQueries);
+        testQueries(json, nonsensicalQueries);
+        //testQueries(json, buggyJaywayQueries);
+
+        for (String errQuery : errQueries) {
+            try {
+                System.out.println("Test failing parse: " + errQuery);
+
+                JsonPathFilterQuery.parse(errQuery);
+                Assert.fail("Parsing the query should have failed: " + errQuery);
+
+            } catch (JsonPathFilterQueryException expected) {
+            }
+        }
+    }
+
+    private void testQueries(JsonNode json, String[] queries) {
+        for (int i = 0; i < queries.length; i++) {
+            String query = queries[i];
+            JsonPathFilterQuery q = JsonPathFilterQuery.parse(query);
+
+            boolean expected = Boolean.parseBoolean(queries[++i]);
+
+            boolean result = q.matches(json);
+            System.out.println("Test: " + query + " " + (result ? "MATCH" : "NO MATCH"));
+
+            Assert.assertEquals("Expected " + expected + " when testing: " + query, expected, result);
+        }
+    }
+
+    @Test
+    public void testPerformance() throws Exception {
```

Comment:
This looks like more of a benchmark than a test: It doesn't look like it can fail. _If_ we need to keep it then perhaps we should rewrite it to work with JMH.

Reply:
Correct, it is a benchmark. It's really just to have a general idea, and I added it in order to compare with my implementation. Using JMH should rather belong to the https://github.com/json-path/JsonPath project, I don't think it makes sense to maintain it here.
I would prefer to either add @Ignore to this test or to completely remove it.

Reply:
Added @Ignore.